src/main/java/sas1/project/demo/model/Speed.java

```diff
@@ -0,0 +1,166 @@
+package sas1.project.demo.model;
+
+public class Speed {
+
+        Ship ship;
+        Wind wind;
+        private int currentSpeed = 0;
+        private int speedIncrement = 0;
+
+        public Speed(Ship ship, Wind wind) {
+            this.ship = ship;
+            this.wind = wind;
+        }
+
+        public int getCurrentSpeed() {
+            return currentSpeed;
+        }
+
+        public void setCurrentSpeed(int currentSpeed) {
+            this.currentSpeed = currentSpeed;
+        }
+
+        public int speedChange(){
+            if (ship instanceof shipBrig){
+                if( ship.getSails() <= 2 && ship.getSails() >=1 ) {
+                    ship.setMaxSpeed(1);
+                }else if(ship.getSails() == 0){
+                    ship.setMaxSpeed(0);
+                }
+            }
+            else if (ship instanceof shipShipOfLine){
+                if (ship.getSails() <= 5 && ship.getSails() >= 3) {
+                    ship.setMaxSpeed(3);
+                }else if(ship.getSails() == 2){
+                    ship.setMaxSpeed(2);
+                }
+                else if(ship.getSails() == 1){
+                    ship.setMaxSpeed(1);
+                }
+                else if(ship.getSails() == 0){
+                    ship.setMaxSpeed(0);
+                }
+            }
+            else if (ship instanceof shipManAtWar){
+                if(ship.getSails() <= 12 && ship.getSails() > 6 ){
+                    ship.setMaxSpeed(3);
+                }else if(ship.getSails() <= 6 && ship.getSails() > 2){
+                    ship.setMaxSpeed(2);
+                }else if(ship.getSails() == 2 || ship.getSails() == 1){
+                    ship.setMaxSpeed(1);
+                }else if (ship.getSails() == 0){
+                    ship.setMaxSpeed(0);
+                }
+            }
+            switch (wind.getWindDirection()){
+                case "N":
+                    if(ship.getDirection().equals(Ship.Direction.N) || ship.getDirection().equals(Ship.Direction.NE) || ship.getDirection().equals(Ship.Direction.NW)){
+                        if(wind.getWindSpeed()>ship.getMaxSpeedChange()){
+                            currentSpeed = ship.getSpeed() + ship.getMaxSpeedChange();
+                        }
+                        else{
+                            currentSpeed = ship.getSpeed() + wind.getWindSpeed();
+                        }
+                    }
+                    else if(ship.getDirection().equals(Ship.Direction.SE) || ship.getDirection().equals(Ship.Direction.SW)){
+                        currentSpeed = ship.getSpeed() -1;
+                    }
+                    else if(ship.getDirection().equals(Ship.Direction.S)){
+                        currentSpeed = 0;
```

Comment:
kan ikke se check på om current speed er indenfor range af hvilke værdi currentspeed max må ændres til.
Hvis curretspeed =5. og den max må gå 2 ned så den bliver 3 istedet for 0 ? 
Måske dette check er et andet sted